File: 1894-merge-strings-alternately/solution.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string result;
        int k=0;
        char* ptr1=&word1[0];
        char* ptr2=&word2[0];
        int in=min(word1.length(),word2.length());
        int incopy=in;
        for(int i=0;i<in;i++) {
            result+=*ptr1++;
            result+=*ptr2++;
        }
        while(in<word1.length())
            result+=word1[in++];
        while(incopy<word2.length())
            result+=word2[incopy++];
        return result;
    }
};

File: 1894-merge-strings-alternately/solution.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        merged = ""
        len1 = len(word1)
        len2 = len(word2)
        min_len = min(len1, len2)
        for i in range(min_len):
            merged += word1[i]
            merged += word2[i]
        return merged + word1[min_len:] + word2[min_len:]
